framework/src/binary_manager/binary_manager_interface.c

```diff
@@ -46,6 +46,15 @@ int binary_manager_set_request(binmgr_request_t *request_msg, int cmd, void *arg
 		}
 		snprintf(request_msg->data.bin_name, BIN_NAME_MAX, "%s", (char *)arg);
 		break;
+	case BINMGR_CREATE_BIN:
+		if (arg == NULL) {
```

Comment:
How about moving arg null check at beginning of the function instead?

Reply:
In case of BINMGR_UNREGISTER_STATECB, it doesn't need arg value.